### PR TITLE
contracts-bedrock: Convert Echidna tests to Forge invariants for Hashing library

### PR DESCRIPTION
**Pull request by Ratimon**
**Description**

I have converted a Echidna test to `Hashing.t.sol` invariant  file.

**Tests**

Please describe any tests you've added. If you've added no tests, or left important behavior untested, please explain why not.

**Additional context**

- https://github.com/ethereum-optimism/ecosystem-contributions/issues/49

**Metadata**

- Fixes #[Convert Echidna tests to Forge invariantes #4980] 

**TODOs**

- [ ] Author or reviewer has added an entry to the [current release notes draft][RND], if appropriate.

[RND]: https://www.notion.so/oplabs/ded30107ceec41c88817e60322aa8d0a?v=b4a22cedb85a46a38c9be14e7c984953&pvs=4 "Release Notes"

### CODE REVIEW

File: packages/contracts-bedrock/contracts/test/invariants/Hashing.t.sol [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    }
+
+    /**
+     * @custom:invariant `hashCrossDomainMessage` reverts if `version` is > `1`.
+     *
+     * The `hashCrossDomainMessage` function should always revert if the `version` passed is > `1`.
+     */
+    function invariant_hash_xdomain_msg_high_version() external {
+        // ASSERTION: The round trip aliasing done in testRoundTrip(...) should never fail.
+        assertEq(actor.failedCrossDomainHashHighVersion(), false);
```
Comment (refcell) [inline packages/contracts-bedrock/contracts/test/invariants/Hashing.t.sol:143]:
```suggestion
        assertFalse(actor.failedCrossDomainHashHighVersion());
```

File: packages/contracts-bedrock/contracts/test/invariants/Hashing.t.sol [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+     * @custom:invariant `version` = `0`: `hashCrossDomainMessage` and `hashCrossDomainMessageV0`
+     * are equivalent.
+     *
+     * If the version passed is 0, `hashCrossDomainMessage` and `hashCrossDomainMessageV0` should be
+     * equivalent.
+     */
+    function invariant_hash_xdomain_msg_0() external {
+        // ASSERTION: A call to hashCrossDomainMessage and hashCrossDomainMessageV0
+        // should always match when the version passed is 0
+        assertEq(actor.failedCrossDomainHashV0(), false);
```
Comment (refcell) [inline packages/contracts-bedrock/contracts/test/invariants/Hashing.t.sol:156]:
```suggestion
        assertFalse(actor.failedCrossDomainHashV0());
```

File: packages/contracts-bedrock/contracts/test/invariants/Hashing.t.sol [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+     * @custom:invariant `version` = `1`: `hashCrossDomainMessage` and `hashCrossDomainMessageV1`
+     * are equivalent.
+     *
+     * If the version passed is 1, `hashCrossDomainMessage` and `hashCrossDomainMessageV1` should be
+     * equivalent.
+     */
+    function invariant_hash_xdomain_msg_1() external {
+        // ASSERTION: A call to hashCrossDomainMessage and hashCrossDomainMessageV1
+        // should always match when the version passed is 1
+        assertEq(actor.failedCrossDomainHashV1(), false);
```
Comment (refcell) [inline packages/contracts-bedrock/contracts/test/invariants/Hashing.t.sol:169]:
```suggestion
        assertFalse(actor.failedCrossDomainHashV1());
```

**Review by refcell — CHANGES_REQUESTED** [review]
Small changes, looks great overall.

It would also be nice to assert the invariant that hashing doesn't revert for valid testing since that won't be caught by these tests if i'm not mistaken. But this can be done in a follow-up pr.

**Review by Ratimon — COMMENTED** [review]
Thanks will PR as suggested

**Pull request merged**